Assets/Scripts/Combat/Abilities/MagicShield.cs

```diff
@@ -0,0 +1,269 @@
+        SequenceTimer.StartTimer(SequenceDuration);
+        CurrentPhase = Phase.SequencePhase;
+          
+    }
+
+    private void SequencePhaseUpdate()
+    {
+
+        if (SequenceTimer.IsInProgress())
+        {
+            if (Sequence.Count != 0)
+            {
+                Button button = Sequence.Dequeue();
+                GameObject direction;
+
+                switch (button)
+                {
+                    case Button.Up:
+                        direction = Instantiate(Arrows[0]);                       
+                        break;
+                    case Button.Down:
+                        direction = Instantiate(Arrows[1]);
+                        break;
+                    case Button.Left:
+                        direction = Instantiate(Arrows[2]);
+                        break;
+                    case Button.Right:
+                        direction = Instantiate(Arrows[3]);
+                        break;
+                    default:
+                        direction = Instantiate(Arrows[0]);
+                        break;
+                }
+
+                direction.transform.localPosition = Position;
+                Position += new Vector3(PositionOffset, 0, 0);
+                direction.SetActive(true);
+                Directions.Enqueue(direction);
+            }
+        }
+
+        if (SequenceTimer.IsFinished())
+        {
+            EndSequencePhase();
+        }
+    }
+
+    private void EndSequencePhase()
+    {
+        foreach (GameObject arrow in Directions)
+            arrow.SetActive(false);
+
+        StartInputPhase();
+    }
+
+    private void StartInputPhase()
+    {
+        SequenceTimer.StartTimer(InputDuration);
+        CurrentPhase = Phase.InputSequence;
+    }
+
+    private void InputSequenceUpdate()
+    {
+        if (SequenceTimer.IsInProgress())
+            CheckUserInputs();
+
+        else if (SequenceTimer.IsFinished())
+            EndInputPhase();
+    }
+
+    private bool IsArrowInputDown()
+    {
+        return Input.GetButtonDown("Up") ||
+            Input.GetButtonDown("Right") ||
+            Input.GetButtonDown("Down") ||
+            Input.GetButtonDown("Left");
+    }
+
+    private void OnCorrectInput()
+    {
+        Debug.Log("OnCorrectInput");
+        
+        CorrectInputs += 1;
+
+        SequenceOrder.Dequeue();
+    }
+
+    private void OnIncorrectInput()
+    {
+        Debug.Log("OnIncorrectInput");
+
+        SequenceOrder.Dequeue();
+    }
+
+    private void CheckUserInputs()
+    {
+        if (!IsArrowInputDown())
+            return;
+
+        Button expectedButton = SequenceOrder.Peek();
+        Debug.Log(expectedButton);
+
+        switch (expectedButton)
+        {
+            case Button.Up:
+                if (Input.GetButtonDown("Right") || Input.GetButtonDown("Down") || Input.GetButtonDown("Left"))
+                    OnIncorrectInput();
+                else
+                    OnCorrectInput();
```

Comment:
Why not check for the correct button first, and then call `OnIncorrectInput()` in the else?

Reply:
Yeah I could do that, would be a lot easier on the eyes too! Daniel added this from his fireball script since its similar. Will fix it now.